leofs: Detect if Application is up issue leo-project/leofs#852

windkit opened the pull request:
## Description
Use nodetool to check the list of running application, when the application is up, the component is ready

## Related to issue
#852

mocchira commented:
@windkit Now the startup script of each component (manager, storage, gateway) is generated(copied when rebar generate) based on https://github.com/leo-project/leofs/blob/master/rel/common/launch.sh a common startup script file so you'd have to modify this file instead of rel/${component}/files/${component} files.
also nodetool seems to be redundant so it's time to move nodetool under rel/common and modify only this file to make the code less redundant and easy to maintain.

windkit commented:
@mocchira OK I can then update the PR to consolidate them and remove the unnecessary files from git

windkit commented:
@mocchira I have updated the PR, merged nodetool into one and updated the launch script.

We also have redundant files under `apps/leo_{gateway,storage,manager}/rel`, should we also link the files up the source tree?

mocchira commented:
@windkit 

> I have updated the PR, merged nodetool into one and updated the launch script.

Thanks, I'll check it.

> We also have redundant files under apps/leo_{gateway,storage,manager}/rel, should we also link the files up the source tree?
Merge 

As other files seem to be remained without any modification for 5 yeas, keep those as is this time.

mocchira commented on rel/common/nodetool:
```diff
@@ -76,6 +79,30 @@ main(Args) ->
+                        _ ->
+                            io:format("Application is not ready\n"),
+                            halt(1)
+                    end
```
http://erlang.org/doc/man/lists.html#any-2 would be more suitable one as it stops to iterate when Predictor func returns true.

```erlang
case lists:any(
fun({leo_gateway,_,_}) ->
    true;
fun({leo_manager,_,_}) ->
    true;
fun({leo_storage,_,_}) ->
    true;
fun(_) ->
    false
end, List) of ->
    true ->
    false ->
end

mocchira commented:
LGTM. Thanks for your contribution.
